platform/core/crates/preferences/src/lib.rs

```diff
@@ -50,3 +52,13 @@ impl TimeoutOptionStore for Preferences {
         self.write(PreferenceKey::UserSessionTimeoutOption, option)
     }
 }
+
+impl UpdateNotificationStore for Preferences {
+    fn get_last_notification(&self) -> Option<LastNotification> {
+        self.read(PreferenceKey::LastNotification)
+    }
+
+    fn set_last_notification(&self, last_notification: LastNotification) {
+        self.write(PreferenceKey::LastNotification, last_notification)
```

Comment:
A case for the nit pickers: I'm somewhat inconsistent with my use of semicolons at the end of blocks. Two rules would make sense to me:

1. No semicolons if the last expression is returned, otherwise semicolon to be clear that it's not a return statement
2. No semicolons at the end of blocks ever, because they're optional

I obviously changed my mind on which way to go a few times :D But I wanna clean that up before I merge this. The existing core code seems inconsistent as well from a few spot checks, but is there any preference?

Reply:
For the case of functions returning Void, probably it would make the most sense to use semicolons.
For the case of functions that return a value, the Rust standard seem to be to not use `return` unless it is required by the code.

Reply:
I looked over my code - generally state inconsistent with the surrounding code, but in new code I added trailing semicolons to void functions.

After this lands, we might want to activate a lint rule for this, found something: https://github.com/rust-lang/rust-clippy/issues/6467